### DISCUSSION TITLE
\lambda update math in PID Lagrangian

### DISCUSSION THREAD

**Discussion by xujinming01** [PKU-Alignment/omnisafe#304] [General]
The docstring in [pid_update()](https://github.com/PKU-Alignment/omnisafe/blob/51a2692afab8afd0cbd2ab2e3d6aec0b79e9af01/omnisafe/common/pid_lagrange.py#L99) shows that the PID controller update the lagrangian multiplier following the next equation:
$\lambda_{t+1} = \lambda_t + (K_p e_p + K_i \int e_p dt + K_d \frac{d e_p}{d t}) \eta$
But I cannot find the \eta in the method, and I don't find where to set it in the cfg.yaml
I think the math should be:
$\lambda_{t+1} = K_p e_p + K_i \int e_p dt + K_d \frac{d e_p}{d t}$
like the original paper:
![Snipaste_2024-02-23_13-18-33](https://github.com/PKU-Alignment/omnisafe/assets/80799351/d7124512-299c-4f38-a043-825ceb021b5e)

I am very much looking forward to your reply.

**Comment by Gaiejj** [comment]
Thanks for your mentioning. Yes, the learning rate $\eta$ mentioned in the document does not have a corresponding interface; this is simply a more generalized expression. Our code implementation is consistent with the original formula provided.
![image](https://github.com/PKU-Alignment/omnisafe/assets/108712610/b04a6810-4a93-4260-b1f3-c5b196545f8f)
We will make the documentation more precise in future updates.
